pandas/core/reshape/reshape.py

```diff
@@ -745,25 +748,19 @@ def _convert_level_number(level_num: int, columns: Index):
             chunk.columns = level_vals_nan.take(chunk.columns.codes[-1])
             value_slice = chunk.reindex(columns=level_vals_used).values
         else:
-            if frame._is_homogeneous_type and is_extension_array_dtype(
-                frame.dtypes.iloc[0]
-            ):
+            subset = this[this.columns[loc]]
```

Comment:
`this.iloc[:, loc]`? i think in case where loc is a slice this avoids a copy

Reply:
updated. thanks